python/packages/kagent-adk/src/kagent/adk/_agent_executor.py

```diff
@@ -262,6 +268,8 @@ async def _handle_request(
                 )
             )
 
+        clear_kagent_span_attributes(context_token)
```

Comment:
Could clearing attributes only here result in keeping stale/invalid attributes for irrelevant spans?

Reply:
Yep, a try/finally block was what I had in mind.